### PR TITLE
Convert apache2.access to ECS - Take 2

### PR DESCRIPTION
**Pull request by webmat**
Taking over #8901, and I don't have write access to @ruflin's repo.

TODO:

- [x] Rebase needed
- [x] Can't alias apache2.access.remote_ip (ecs-migration.yml)
- [x] Create aliases in the module's field definitions
- [x] HTTP metrics comment (not a deal breaker)
- [x] source.hostname not in ECS
- [x] referer typo
- [x] user_agent parsing currently broken
- [x] Add test for a named host as the remote
- [x] Make integer fields great again
- [x] Changelog
- [x] Fix rebase issues
- [x] Remove commented out `url.original` from `filebeat/_meta/fields.common.yml`. It's a dupe of `libbeat/_meta/fields.ecs.yml` definition.
- [x] Figure out apparently [unrelated breakage](https://travis-ci.org/elastic/beats/jobs/460918667) about a generated file: `https://travis-ci.org/elastic/beats/jobs/460918667` 

Unrelated

- [ ] Did I list the geoip aliases in ecs-migration.yml in the other PRs?

Closes #8901, partially addresses #8655.

### CODE REVIEW

**Comment by webmat** [comment]
@ruflin I brought your initial Apache2 access PR exactly in line with the other recent access log PRs. Expecting to have the build work the first time.

File: auditbeat/docs/fields.asciidoc [hunk 1]
```diff
@@ -4387,10 +4387,10 @@ URL fields provide a complete URL, with scheme, host, and path. The URL object c
 
 
 
-*`url.href`*::
+*`url.original`*::
```
[the remark this answers is not in the document]
Comment (webmat) [reply auditbeat/docs/fields.asciidoc]: I've been using `url.original` already in many of the merged PRs. If there was an issue, it's been handled last week :-)

**Comment by webmat** [comment]
Only failure in Jenkins was a network hiccup [on Ubuntu](https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/beat=auditbeat,label=ubuntu/1994/console):

`22:30:27 gvm: error: http get failed: Get https://storage.googleapis.com/golang/go1.11.2.linux-amd64.tar.gz: dial tcp 108.177.112.128:443: i/o timeout`

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -132,6 +132,8 @@ https://github.com/elastic/beats/compare/v6.5.0...v7.0.0-alpha1[View commits]
 - Rename many `haproxy.*` fields to map to ECS. {pull}9117[9117]
 - Rename many `nginx.access.*` fields to map to ECS. {pull}9081[9081]
 - Rename many `system.auth.*` fields to map to ECS. {pull}9138[9138]
+- Rename many `apache2.access.*` fields to map to ECS. {pull}8901[8901]
+- Rename many `apache2.access.*` fields to map to ECS. {pull}9245[9245]
```
[the remark this answers is not in the document]
Comment (webmat) [reply CHANGELOG.asciidoc]: Ugh. Another rebase mixup. Will pore over my diff again and fix this one specifically.

[the remark this answers is not in the document]
Comment (webmat) [reply CHANGELOG.asciidoc]: Thanks for catching that. Should be good to go, now.

**Comment by webmat** [comment]
@ruflin Ok, ready for final review. Everything is green, and even fixes the [test breakage in master](https://github.com/elastic/beats/pull/9245/commits/462c9e638a2247b1dcf336a4d0ef8e48a424447f).

File: filebeat/docs/fields.asciidoc [hunk 3]
```diff
@@ -64,270 +64,183 @@ Contains fields for the Apache2 HTTPD access logs.
 --
 type: keyword
 
-Client IP address.
+Client IP address or hostname.
```
Comment (jsoriano) [inline filebeat/docs/fields.asciidoc]: Just curiosity, can the client host be logged as a hostname by apache?

File: filebeat/docs/fields.asciidoc [hunk 4]
```diff
@@ -64,270 +64,183 @@ Contains fields for the Apache2 HTTPD access logs.
...
+*`user.name`*::
 +
 --
-type: keyword
-
-example: GET
-
-The request HTTP method.
-
+type: alias
```
Comment (jsoriano) [inline filebeat/docs/fields.asciidoc]: Is this what will appear in the documentation on aliases? We should probably take the description of the original field or something like this.

Not for this PR in any case as this file is generated.

**Comment by jsoriano** [comment]
For a future change, we might consider to rename the `apache2` module to `apache`.

**Comment by webmat** [comment]
@jsoriano Actually "Apache" is the foundation. The webserver is actually called "httpd" ;-)

**Comment by webmat** [comment]
But I totally agree on another part of your point. The number is problematic. Will "apache2" support Apache httpd v3? So yeah, the number has to go eventually

File: filebeat/docs/fields.asciidoc (hunk 3 above)
Comment (webmat) [reply filebeat/docs/fields.asciidoc]: Yes, the position of  `%h` in the logs can be a hostname if the option `HostnameLookups` is set to `On`

Comment (webmat) [reply filebeat/docs/fields.asciidoc]: (default is Off)

File: filebeat/docs/fields.asciidoc (hunk 4 above)
Comment (webmat) [reply filebeat/docs/fields.asciidoc]: Documentation on aliases is being worked on generically in #9269, don't worry. I agree this looks terrible ;-)

**Comment by jsoriano** [comment]
@webmat apart of the number I was mentioning that to use the same name as the metricbeat module, though we can also rename both to `httpd` :smile:

File: filebeat/docs/fields.asciidoc (hunk 3 above)
Comment (jsoriano) [reply filebeat/docs/fields.asciidoc]: Oh interesting :+1:

**Comment by webmat** [comment]
Ah I was not aware of that. Thanks for mentioning it! `>> /mat/brain` 😆

**Pull request merged**